packages/harmony/src/components/expandable-nav-item/ExpandableNavItem.tsx

```diff
@@ -0,0 +1,114 @@
+import { useMemo, useState } from 'react'
+
+import { useTheme, CSSObject } from '@emotion/react'
+
+import { HarmonyTheme } from '../../foundations/theme'
+import { IconCaretDown, IconCaretRight } from '../../icons'
+import { Flex } from '../layout/Flex'
+import { Text } from '../text'
+
+import type { ExpandableNavItemProps } from './types'
+
+const getStyles = (
+  theme: HarmonyTheme,
+  isOpen: boolean,
+  isHovered: boolean
+): CSSObject => {
+  const baseStyles: CSSObject = {
+    transition: `background-color ${theme.motion.hover}`,
+    cursor: 'pointer',
+    border: `1px solid ${isOpen ? theme.color.border.default : 'transparent'}`
+  }
+
+  const hoverStyles: CSSObject = {
+    backgroundColor: theme.color.background.surface2
+  }
+
+  return {
+    ...baseStyles,
+    ...(isHovered && hoverStyles)
+  }
+}
+
+export const ExpandableNavItem = ({
+  label,
+  leftIcon: LeftIcon,
+  rightIcon,
+  defaultIsOpen = false,
+  nestedItems,
+  shouldPersistRightIcon = false,
+  ...props
+}: ExpandableNavItemProps) => {
+  const [isOpen, setIsOpen] = useState(defaultIsOpen)
+  const [isHovered, setIsHovered] = useState(false)
+  const theme = useTheme()
+
+  const handleMouseEnter = () => setIsHovered(true)
+  const handleMouseLeave = () => setIsHovered(false)
+  const handleClick = () => setIsOpen(!isOpen)
+
+  const styles = useMemo(
+    () => getStyles(theme, isOpen, isHovered),
+    [theme, isOpen, isHovered]
+  )
+
+  const IconComponent = isHovered
+    ? isOpen
+      ? IconCaretDown
+      : IconCaretRight
+    : LeftIcon
+
+  return (
+    <Flex direction='column' {...props}>
+      <Flex
+        alignItems='center'
+        gap='s'
+        pl='s'
+        pr='s'
+        css={{
+          width: '240px',
+          cursor: 'pointer',
+          transition: `background-color ${theme.motion.hover}`
+        }}
+      >
+        <Flex
+          alignItems='center'
+          flex={1}
+          gap='m'
+          p='s'
+          borderRadius='m'
+          css={styles}
+          onMouseEnter={handleMouseEnter}
+          onMouseLeave={handleMouseLeave}
+          onClick={handleClick}
+        >
+          <Flex
+            alignItems='center'
+            gap='m'
+            flex={1}
+            css={{
```

Comment:
it does feel like the "nav-item" aspect of this makes it more a "web" component than a harmony one. what do you think? there is a world where the harmony component is "ExpandablePanel" that could be used in a bunch of places, including implementing the "ExpandableNavItem"? I don't think we need to over-engineer too much here though and can just have an expandable-nav-item, but perhaps that just lives in web?

Reply:
we also want to make sure the nav elements accessible role is connected to the label, which may just happen automatically, but just want to make sure that happens. hmu if you need some help getting set up here.